psh/ping/ping.c

```diff
@@ -260,25 +422,167 @@ static int ping_resolveAddress(int af, const char *src, void *dst)
 	return (rp == NULL) ? -1 : 0;
 }
 
-
-int psh_ping(int argc, char **argv)
+static int ping_sendping4(const char *src)
 {
 	char addrstr[INET_ADDRSTRLEN];
 	uint8_t *resp, *req;
-	int fd, c, ret = 0;
+	int fd;
+	int ret = 0;
+
+	ping_common.af = AF_INET;
+	ping_common.raddr.sin_family = AF_INET;
+	ping_common.respsz = sizeof(struct iphdr) + ping_common.reqsz;
+
+	if (ping_resolveAddress(ping_common.af, src, &ping_common.raddr.sin_addr) != 0) {
+		fprintf(stderr, "ping: cannot resolve address: %s\n", src);
+		return 2;
+	}
+
+	if (inet_ntop(ping_common.af, &ping_common.raddr.sin_addr, addrstr, sizeof(addrstr)) == NULL) {
+		fprintf(stderr, "ping: Invalid address\n");
+		return 2;
+	}
+
+	if ((req = malloc(ping_common.reqsz)) == NULL) {
```

Comment:
Follow MISRA here and in other places:


req = malloc...;
if (req == NULL) {
}